sim/mage/talents.go

```diff
@@ -122,15 +122,29 @@ func (mage *Mage) ApplyTalents() {
 	/* --------------------------------------
 				  FIRE TALENTS
 	---------------------------------------*/
-	fireMastery := mage.AddDynamicMod(core.SpellModConfig{
-		ClassMask:  MageSpellFireDoT,
-		FloatValue: float64(1.22 + 0.28*mage.GetMasteryPoints()),
-		Kind:       core.SpellMod_DamageDone_Pct,
-	})
+	// Fire  Specialization Bonus
+	if mage.Spec == proto.Spec_SpecFireMage {
+		mage.AddStaticMod(core.SpellModConfig{
+			School:     core.SpellSchoolFire,
+			FloatValue: 0.25,
+			Kind:       core.SpellMod_DamageDone_Flat,
+		})
+	}
 
-	mage.AddOnMasteryStatChanged(func(sim *core.Simulation, oldMastery, newMastery float64) {
-		fireMastery.UpdateFloatValue(1.22 + 0.28*core.MasteryRatingToMasteryPoints(newMastery))
-	})
+	// Mastery
+	if mage.Spec == proto.Spec_SpecFireMage {
+		//
+		fireMastery := mage.AddDynamicMod(core.SpellModConfig{
+			ClassMask:  MageSpellFireDoT,
+			FloatValue: float64(1.22 + 0.28*mage.GetMasteryPoints()),
+			Kind:       core.SpellMod_DamageDone_Pct,
+		})
+		fireMastery.Activate()
+
+		mage.AddOnMasteryStatChanged(func(sim *core.Simulation, oldMastery, newMastery float64) {
+			fireMastery.UpdateFloatValue(1.22 + 0.28*core.MasteryRatingToMasteryPoints(newMastery))
+		})
+	}
```

Comment:
This is leaking spec specific code in the class package. Instead you should move this to the fire specs go files. Check DK for example setup

Reply:
Also update should be `0.22 + ...` insted of `1.22` (The spell mod already adds 1)